common/beerocks/hostapd/configuration.cpp

```diff
@@ -0,0 +1,390 @@
+            line.erase(0, end_comment);
+        };
+    });
+}
+
+const std::string &Configuration::get_last_message() const { return m_last_message; }
+
+std::tuple<bool, std::vector<std::string> *>
+Configuration::get_vap(const std::string &calling_function, const std::string &vap)
+{
+    // search for the requested vap - ignore comments
+    // by searching from the back of the saved vap (rfind)
+    auto existing_vap =
+        std::find_if(m_hostapd_config_vaps.begin(), m_hostapd_config_vaps.end(),
+                     [&vap](const std::pair<std::string, std::vector<std::string>> &current_vap) {
+                         return current_vap.first.rfind(vap) != std::string::npos;
+                     });
+
+    if (existing_vap == m_hostapd_config_vaps.end()) {
+        m_last_message = calling_function + " couldn't find requested vap: " + vap;
+        m_ok           = false;
+        return std::make_tuple(false, nullptr);
+    }
+
+    m_ok = true;
+    return std::make_tuple(true, &existing_vap->second);
+}
+
+bool Configuration::is_key_in_line(const std::string &line, const std::string &key) const
+{
+    // we need to make sure when searching for example
+    // for "ssid", to ignore cases like:
+    // multi_ap_backhaul_ssid="Multi-AP-24G-2"
+    //                   ^^^^^
+    // bssid=02:9A:96:FB:59:11
+    //  ^^^^^
+    // and we need to take into consideration
+    // that the key might be or might not be commented.
+    // so the search algorithm is:
+    // - find the requested key and
+    // - make sure it is either on the first position
+    // or it has a comment sign just before it
+    auto found_pos = line.rfind(key);
+    bool ret = found_pos != std::string::npos && (found_pos == 0 || line.at(found_pos - 1) == '#');
+
+    return ret;
+}
+
+std::ostream &operator<<(std::ostream &o, const Configuration &conf)
+{
+    o << "== configuration details ==\n"
+      << "= ok:           " << std::boolalpha << conf.m_ok << '\n'
+      << "= last message: " << conf.m_last_message << '\n'
+      << "= file:         " << conf.m_configuration_file << '\n'
+      << "= head:         " << '\n';
+
+    for (const auto &line : conf.m_hostapd_config_head) {
+        o << line << '\n';
+    }
+
+    o << "== vaps (total of: " << conf.m_hostapd_config_vaps.size() << " vaps) ==\n";
+
+    for (const auto &vap : conf.m_hostapd_config_vaps) {
+        o << "   vap: " << vap.first << "\n";
+        for (const auto &line : vap.second) {
+            o << line << '\n';
+        }
+    }
+
+    return o;
+}
+
+} // namespace hostapd
+} // namespace prplmesh
+
+/*
+//////////////// hostapd configuration format ////////////////
+
+hostapd has a special format that does NOT have an ini like format.
+between /// BEGIN hostapd.conf /// and /// END hostapd.conf /// is the 
+format of the file.
+note: the string "bss=" may be replaced by the
+user in the call to load() with another vap-indicator (e.g. "interface=")
```

Comment:
This is wrong. The actual format of the file is:
```
... Configuration of first VAP
bss=...
... Configuration of second VAP
bss=...
... Configuration of third VAP
```
`interface=` sets the name of the VAP. `bss=` also sets it, so it really is only needed for the first VAP.

As it happens, openwrt generates the configuration file as:
```
... Global options
interface=first_vap_name
... First VAP options
bss=second_vap_name
... Second VAP options
```

But that's just "coincidence", not really something we can rely on...